tests/testthat/test-shinytest2-teal-widgets.R

```diff
@@ -0,0 +1,218 @@
+          label = "Module with table_with_settings",
+          ui = function(id) {
+            teal.widgets::table_with_settings_ui(id = id)
+          },
+          server <- function(id, input, output, session) {
+            table_r <- reactive({
+              layout <- rtables::basic_table() %>%
+                rtables::split_cols_by("ARM") %>%
+                rtables::analyze(c("SEX", "AGE"))
+
+              rtables::build_table(layout, formatters::DM)
+            })
+
+            teal.widgets::table_with_settings_srv(id = id, table_r = table_r)
+          }
+        )
+      )
+    )
+
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Check if there are two buttons above the table.
+    table_buttons_selector <- app$active_module_element("table-with-settings > div.table-settings-buttons")
+    table_buttons <-
+      app$get_html_rvest(table_buttons_selector) %>%
+      rvest::html_elements("button")
+    testthat::expect_length(table_buttons, 2)
+    # Check is the first one is a toggle button.
+    testthat::expect_equal(
+      table_buttons[[1]] %>%
+        html_attr("data-toggle"),
+      "dropdown"
+    )
+    # First button has specific font-awesome icon.
+    dwnl_button <- app$active_module_element("downbutton-dwnl")
+    testthat::expect_equal(
+      app$get_html_rvest(dwnl_button) %>%
+        rvest::html_element("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-download"
+    )
+
+    # Click the first TABLE button.
+    app$click(selector = dwnl_button)
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Review the content of the toggle.
+    testthat::expect_equal(
+      app$active_module_element_text("downbutton-file_format-label"),
+      "File type"
+    )
+
+    file_format_text <- app$active_module_element_text("downbutton-file_format > div")
+    testthat::expect_match(file_format_text, "formatted txt\n", fixed = TRUE)
+    testthat::expect_match(file_format_text, "csv\n", fixed = TRUE)
+    testthat::expect_match(file_format_text, "pdf\n", fixed = TRUE)
+
+    testthat::expect_equal(
+      app$active_module_element_text("downbutton-file_name-label"),
+      "File name (without extension)"
+    )
+
+    # QUESTION - NEED HELP
+    # How to get
+    # value="table_20240321_142134"
+    # out of
+    # <input id="teal-main_ui-root-module_with_table_with_settings-module-downbutton-file_name" type="text" class="shiny-input-text form-control shinyjs-resettable shiny-bound-input" value="table_20240321_142134" data-shinyjs-resettable-id="teal-main_ui-root-module_with_table_with_settings-module-downbutton-file_name" data-shinyjs-resettable-type="Text" data-shinyjs-resettable-value="table_20240321_142134">
+    # for
+    # app$get_html_rvest(app$active_module_element("downbutton-file_name"))
+
+    # TODO - NEED HELP
+    # do not use div:nth-child(3)
+    pagination <- "downbutton-dwnl > li > div > div:nth-child(3) > div.paginate-ui  > div.form-group.shiny-input-container"
```

Comment:
There's only 1 `.paginate-ui`

I'd say that `downbutton-dwnl .paginate-ui .form-group.shiny-input-container` is specific.

I believe that we don't need to explicitly define the whole path and can work with mostly classes here.

Reply:
Thanks. I just couldn't get this simplified - now I see how I can work it out in the future!

Reply:
thanks - simplified in here f002f46a7e84071be9a410b5ab641eeb972ba33b